pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -1261,8 +1262,13 @@ def get_maximum_resources_available(self, chips=None, processor_id=None,
                 best_processor_id = self._best_core_available(chip, key,
                                                               processor_id)
                 processor = chip.get_processor_with_id(best_processor_id)
-                max_dtcm_available = processor.dtcm_available
-                max_cpu_available = processor.cpu_cycles_available
+                try:
+                    max_dtcm_available = processor.dtcm_available
+                    max_cpu_available = processor.cpu_cycles_available
+                except AttributeError:
```

Comment:
That's very pythonic... would it be better to do the processor is None check afterwards?

Reply:
I this case NO. In all normally conditions None is not expected.
Therefor the code reads correctly in this should not happen but in case of emergency do....

I had wondered why chip.get_processor_with_id actually return None but as it specifically does assumed there was a valid use case so did not change that.